### PR TITLE
Support padding for collation

### PR DESCRIPTION
**Pull request by yycho0108**
## Description

Previously batching was not possible for `>1` instances in a frame due to inconsistent dimensions in some axes.
This PR adds support for batching objectron-like datasets. Follow-up PR will address collation function for cropped images.

## Changelog
* rename objectron datasets for disambiguation
* parallel loading support in detection objectron
* fix cached dataset loading in keypoint regression
* fix tensor shape inconsistencies regarding instance num
* use `gcs` + `tfrecord` instead of `torch_xla` everywhere
* add missing feature keys to objectron dataset
* add configurable instance-padding transform
* add collation loading test script
* deprecate `gcs` arg on tfrecord loaders wrt upstream change
* split `transforms` into multiple files
* modify number of keypoints computed from displacement map
* refactor out common data loader

## Test Collation
```
$ python3 test_collate.py 
Schema.IMAGE torch.Size([8, 3, 640, 480])
Schema.KEYPOINT_2D torch.Size([8, 4, 9, 3])
Schema.INSTANCE_NUM torch.Size([8])
Schema.TRANSLATION torch.Size([8, 4, 3])
Schema.ORIENTATION torch.Size([8, 4, 9])
Schema.SCALE torch.Size([8, 4, 3])
Schema.PROJECTION torch.Size([8, 16])
Schema.KEYPOINT_NUM torch.Size([8, 4])
Schema.CLASS torch.Size([8, 4])
Schema.HEATMAP torch.Size([8, 9, 160, 120])
Schema.DISPLACEMENT_MAP torch.Size([8, 18, 160, 120])
```

@ajy8456 In order to test this PR you need to update your `tfrecord` package version:
`poetry update tfrecord` (takes about 2 minutes due to very slow dependency resolution for some reason)

### CODE REVIEW

File: scripts/bb_regression.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
 from top.model.bbox_3d import BoundingBoxRegressionModel
 from top.model.loss_util import orientation_loss, generate_bins
 
-from top.data.objectron_dataset_detection import Objectron
+from top.data.objectron_detection import Objectron
```
Comment (yycho0108) [inline scripts/bb_regression.py:27]: changes in this file except for this are autoformatting artifacts, feel free to ignore.

**Comment by yycho0108** [comment]
There are a number of changes that were added to this PR despite belonging to a different effort regarding improving the training of the keypoint regressor. Feel free to either review those changes or table it to a later PR. @ajy8456

File: src/top/data/transforms/common.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    def __call__(self, inputs: Dict[Schema, th.Tensor]):
+        outputs = inputs.copy()
+        for k in self.opts.pad_keys:
+            if k not in inputs:
+                continue
+            pad = [0] * (2 * len(inputs[k].shape))
+            pad[-2] = int(self.opts.max_num_inst - inputs[Schema.INSTANCE_NUM])
+            x = th.as_tensor(inputs[k])
+            outputs[k] = F.pad(x, pad)
```
Comment (ajy8456) [inline src/top/data/transforms/common.py:94]: I understand that the meaning of `pad[-2]` is to pad the batch dimension as much as the difference between `self.opts.max_num_inst` and `Schema.INSTANCE_NUM`. Is it correct?

**Review by ajy8456 — APPROVED** [review]
LGTM~!

File: src/top/data/transforms/common.py (hunk 2 above)
Comment (yycho0108) [reply src/top/data/transforms/common.py:94]: yes.

**Pull request merged**